### ISSUE TITLE
Input 0: [2,32,32,512] != input 1: [1048576]

### ISSUE THREAD

**Issue by ZhengQingWangg** [zhixuhao/unet#96]
InvalidArgumentError (see above for traceback): Inputs to operation training_2/Adam/gradients/AddN_3 of type _MklAddN must have the same size and shape.  Input 0: [2,32,32,512] != input 1: [1048576]
	 [[Node: training_2/Adam/gradients/AddN_3 = _MklAddN[N=2, T=DT_FLOAT, _kernel="MklOp", _device="/job:localhost/replica:0/task:0/device:CPU:0"](training_2/Adam/gradients/merge_5/concat_grad/Slice, training_2/Adam/gradients/max_pooling2d_8/MaxPool_grad/MaxPoolGrad, DMT/_437, training_2/Adam/gradients/max_pooling2d_8/MaxPool_grad/MaxPoolGrad:1)]]

**Comment by virus1987** [comment]
i have the same questions,please help me ~~

**Comment by virus1987** [comment]
@ZhengQingWangg did u solved this issue?
@zhixuhao my code is run on cpu with tensorflow ver = 1.9.0 and keras ver = 2.1.0 
did this problem above metioned occur while the versions are not compatible ?
i am wating for the replay 3q

**Comment by ZhengQingWangg** [comment]
> @ZhengQingWangg你解决了这个问题吗？
> @zhixuhao我的代码在cpu上运行，张量为ver = 1.9.0，而keras ver = 2.1.0 
> 上述问题是否在版本不兼容时出现？
> 我正在为重播3q做好准备

I solved my problem, just change the version of python.

**Comment by ZhengQingWangg** [comment]
> 我有同样的问题，请帮帮我~~

My original version is 3.7, I solved the problem by changing the version to 3.6.

**Comment by virus1987** [comment]
@ZhengQingWangg thanks a lot~i solved the problem as your solution~

**Comment by hzhenhuan** [comment]
@ZhengQingWangg @virus1987  I  met same questions, too.my code is run on cpu with tensorflow ver = 1.9.0 and keras ver = 2.2.4，python ver = 3.6 ，please help me ,tks

**Comment by LeslieDang** [comment]
好吧~我用的python3.7，也遇到了同样的问题~查了老半天才在这里看到，竟然是python版本问题！！！吐一口老血~

**Comment by spyitachi** [comment]
My original version is 3.6, but it still came up

**Comment by spyitachi** [comment]
Upgrade Tensorflow and Keras by pip3 on 3.6 or 3.7 version!
